src/azure-cli/azure/cli/command_modules/acr/_format.py

```diff
@@ -93,6 +93,55 @@ def agentpool_output_format(result):
     return _output_format(result, _agentpool_format_group)
 
 
+def connected_registry_output_format(result):
+    return _output_format(result, _connected_registry_format_group)
+
+
+def connected_registry_list_output_format(result):
+    family_tree = {}
+    for reg in result:
+        parent_id = _get_value(reg, 'parent', 'id')
+        parent_name = '' if parent_id.isspace() else parent_id.split('/connectedRegistries/')[1]
+        name = ' ' + _get_value(reg, 'name')
+        family_tree[_get_value(reg, 'id')] = {
+            "name": name,
+            "id": _get_value(reg, 'id'),
+            "connectionState": _get_value(reg, 'connectionState'),
+            "parent_name": parent_name,
+            "parent_id": parent_id,
+            "loginServer_host": _get_value(reg, 'loginServer', 'host'),
+            "parent_syncProperties_lastSyncTime": _get_value(reg, 'parent', 'syncProperties', 'lastSyncTime'),
+            "mode": _get_value(reg, 'mode'),
+            "childs": []
+        }
+
+    parents = []
+    for reg in result:
+        parent_id = _get_value(reg, 'parent', 'id')
+        if parent_id.isspace() or parent_id not in family_tree:
```

Comment:
curious as to what this check as not in family_tree

Reply:
so this is for keeping the tree format when the user uses the --parent option. Since the output was filter to only the children of the parent connected registry given, the resources that doesn't have a parent in the family_tree would be the level 0 resources of the tree.